Message_changed message payload does not have a `text` property

```
May 9 12:19:40 PM  [ERROR]  bolt-app UnknownError: Cannot read property 'replace' of undefined
May 9 12:19:40 PM      at asCodedError (/opt/render/project/src/node_modules/@slack/bolt/dist/errors.js:35:12)
May 9 12:19:40 PM      at App.handleError (/opt/render/project/src/node_modules/@slack/bolt/dist/App.js:607:57)
May 9 12:19:40 PM      at App.processEvent (/opt/render/project/src/node_modules/@slack/bolt/dist/App.js:592:25)
May 9 12:19:40 PM      at processTicksAndRejections (internal/process/task_queues.js:95:5)
May 9 12:19:40 PM      at async /opt/render/project/src/node_modules/@slack/bolt/dist/receivers/HTTPReceiver.js:313:17 {
May 9 12:19:40 PM    code: 'slack_bolt_unknown_error',
May 9 12:19:40 PM    original: TypeError: Cannot read property 'replace' of undefined
May 9 12:19:40 PM        at Array.module.exports (/opt/render/project/src/listenerMiddleware/fpAsMention.js:2:33)
May 9 12:19:40 PM        at invokeMiddleware (/opt/render/project/src/node_modules/@slack/bolt/dist/middleware/process.js:12:53)
May 9 12:19:40 PM        at next (/opt/render/project/src/node_modules/@slack/bolt/dist/middleware/process.js:13:29)
May 9 12:19:40 PM        at Array.<anonymous> (/opt/render/project/src/node_modules/@slack/bolt/dist/middleware/builtin.js:245:15)
May 9 12:19:40 PM        at invokeMiddleware (/opt/render/project/src/node_modules/@slack/bolt/dist/middleware/process.js:12:53)
May 9 12:19:40 PM        at next (/opt/render/project/src/node_modules/@slack/bolt/dist/middleware/process.js:13:29)
May 9 12:19:40 PM        at Array.onlyEvents (/opt/render/project/src/node_modules/@slack/bolt/dist/middleware/builtin.js:73:11)
May 9 12:19:40 PM        at invokeMiddleware (/opt/render/project/src/node_modules/@slack/bolt/dist/middleware/process.js:12:53)
May 9 12:19:40 PM        at processMiddleware (/opt/render/project/src/node_modules/@slack/bolt/dist/middleware/process.js:22:12)
May 9 12:19:40 PM        at /opt/render/project/src/node_modules/@slack/bolt/dist/App.js:566:50
May 9 12:19:40 PM  }
```

--- a/listenerMiddleware/__tests__/fpAsMention.test.js
+++ b/listenerMiddleware/__tests__/fpAsMention.test.js
@@ -1,0 +1,46 @@
+const fpAsMention = require('../fpAsMention');
+
+test('fp in message text gets converted to a user id', async () => {
+    request = {
+        message: {
+            text: 'fp'
+        },
+        context: {
+            botUserId: 'botuserid'
+        },
+        next: jest.fn()
+    };
+
+    await fpAsMention(request);
+
+    expect(request.message.text).toBe('<@botuserid>');
+});
+
+test('no fp in message text does not change the message text', async () => {
+    request = {
+        message: {
+            text: 'test text'
+        },
+        context: {
+            botUserId: 'botuserid'
+        },
+        next: jest.fn()
+    };
+
+    await fpAsMention(request);
+
+    expect(request.message.text).toBe('test text');
+});
+
+test('no message text does not throw an error', async () => {
+    request = {
+        message: {
+
+        },
+        next: jest.fn()
+    };
+
+    await fpAsMention(request);
+
+    expect(true).toBe(true);
+});

--- a/listenerMiddleware/fpAsMention.js
+++ b/listenerMiddleware/fpAsMention.js
@@ -1,4 +1,8 @@
 module.exports = async ({message, context, next}) => {
+    if (!message.text) {
+        return await next();
+    }
+
     message.text = message.text.replace(/\bfp\b/, `<@${context.botUserId}>`);
 
     await next();

